CIP-1775/README.md

```diff
@@ -0,0 +1,108 @@
+---
+CIP: 1775
+Title: Delegated Spending Authorizations
+Authors: Ivan Mikushin <sorrel_cantos.0@icloud.com>
+Comments-URI:
+Status: Proposed
+Type: Standards Track
+Created: 2022-11-17
+License: CC-BY-4.0
+---
+
+# Delegated Spending Authorizations
+
+"Give me custody, or give me death!" - [@ArmySpies](https://twitter.com/ArmySpies/status/1592586212816355328)
+
+## Abstract
+
+We propose to allow users to delegate authorized spending of outputs in their wallets to smart contracts or other users, without sending said outputs out of their wallets.
+
+## Motivation
+
+In order to use any dApp on Cardano (e.g. a decentralized exchange or an NFT marketplace), a user has to **deposit tokens** to a script address, thereby **giving up custody** of the tokens.
+
+This can be problematic for a number of reasons.
+
+- The user loses control of the assets they send out to a smart-contract, unless the smart contract is specifically designed to allow for such control (e.g. giving a "Cancel trade" or "De-list NFT" button in the UI).
+- Assets sent out from a user's wallet:
+  - are no longer  **delegated** to the stake pool of the user's choice.
+  - cannot participate in **governance voting** on behalf of the user.
+- Any output, accidentally sent to a script address without a datum is effectively lost now (because of how transaction validation scripts are invoked).
+- Requiring users to deposit assets into a dapp may subject the dapp's owners to regulatory scrutiny.
+
+We believe, this is not strictly necessary and would like a way to allow users to delegate spending authorization for some outputs **within** their wallets to other users (smart contracts, multisigs, wallet addresses).
+
+### Use Cases
+
+Instead of sending tokens to the dapp smart-contract, the user creates an output in their own wallet that delegates spending authorization to a smart-contract (or other entity).
+
+Some **existing** use cases, where this can be beneficial:
+
+1. Swapping tokens on a decentralized exchange.
+2. Providing liquidity on a decentralized exchange.
```

Comment:
How does the DEX monitor these thousands of delegated utxos to ensure they haven't been spent yet? In the current case where the DEX has custody of the funds, they only have to monitor 1 or 2 UTxOs to know what the state of the liquidity pool is. In the case of thousands of UTxOs to monitor, the smart contract becomes un-runnable due to size. 

If there's no common liquidity pool to swap with, how do you take just a sliver of tokens from thousands of utxos without making them no longer spendable. The whole concept of a DEX doesn't work unless custody is given. Without custody, there's no guaranteed-available liquidity there.

Reply:
The point isn't to take away full custody from DApps.

The point is to introduce an additional, non-custodial, way to interact with DApps that support it.

> How does the DEX monitor these thousands of delegated utxos to ensure they haven't been spent yet? In the current case where the DEX has custody of the funds, they only have to monitor 1 or 2 UTxOs to know what the state of the liquidity pool is. 

Can't they run an indexer off-chain to know the size? I didn't know that a liquidity pool can only be 1 or 2 UTXOs. (And still don't know 🙂). If I'm one of thousands of people providing liquidity, how can I withdraw my assets? My ownership has to be encoded in these 1 or 2 UTXOs. No wonder people complain about concurrency. It doesn't have to be **only** this way, though.

> In the case of thousands of UTxOs to monitor, the smart contract becomes un-runnable due to size.

Aren't smart-contracts in Cardano designed to validate spending of a **single** output? Which one to spend can be decided off-chain, I imagine.

> If there's no common liquidity pool to swap with, how do you take just a sliver of tokens from thousands of utxos without making them no longer spendable. 

I'd imagine, an off-chain process matches swap orders with these thousands of UTXOs and constructs transactions. A transaction for each swap order, spends one or more (just a few, not thousands) such UTXOs. UTXOs with the remaining and incoming tokens are created back in the LP users' wallets.

If, by the tx execution time, some of the LP UTXOs are already spent, the transaction is re-constructed, until executed. This is classic [Optimistic concurrency control](https://en.wikipedia.org/wiki/Optimistic_concurrency_control).

> The whole concept of a DEX doesn't work unless custody is given. Without custody, there's no guaranteed-available liquidity there.

Nothing in this world is guaranteed (except for taxes and death, of course). But! Optimistic concurrency does work, and I don't see why not make use of it for the benefit of Cardano users.